MatrixBuilder.cpp

```diff
@@ -0,0 +1,293 @@
+
+#include "MatrixBuilder.h"
+
+// Build a new Matrix by combining two Matrices according to specification.
+// @param const Matrix& first - the first Matrix involved in the combination
+// @param const Matrix& last - the second Matrix involved in the combination
+// @param MatrixCombinatoinType rule - specify how the Matrices will be combined:
+//									   (1) LeftRight
+//									   (2) TopDown
+// @return - a new Matrix combined from the other two
+// @throws exceptions for unsupported MatrixCombinationType and mismatched dimensions.
+Matrix MatrixBuilder::BuildMatrixFromCombination(
+		const Matrix& first, const Matrix& last, MatrixCombinationType rule) {
+	switch (rule) {
+		case LeftRight:
+			if (first.size(1) != last.size(1)) {
+				throw "ERROR  "
+				  "Matrix MatrixBuilder::BuildMatrixFromCombination(const Matrix&, const Matrix&, MatrixCombinationType)\n"
+				  "\tInpt Matrices must have the same number of rows for LeftRight Matrix Combination";
+			} else {
+				int nRows = first.size(1);
+				int nColsLeft = first.size(2);
+				int nColsRight = last.size(2);
+				vector<vector<double>> helper(nRows, vector<double>(nColsLeft + nColsRight));

```

Comment:
Why use these helper vectors? Why not just construct the Matrix of appropriate size and fill it in instead of the vectors?